pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -339,4 +306,22 @@ impl InMemoryLayer {
         let delta_layer = delta_layer_writer.finish(Key::MAX)?;
         Ok(delta_layer)
     }
+
+    pub fn get_current_lsn_range(&self) -> Range<Lsn> {
+        let inner = self.inner.read().unwrap();
+
+        let end_lsn = if let Some(end_lsn) = inner.end_lsn {
+            end_lsn
+        } else {
+            Lsn(u64::MAX)
+        };
+        self.start_lsn..end_lsn
+    }
+
+    pub fn short_id(&self) -> String {
+        let inner = self.inner.read().unwrap();
+
+        let end_lsn = inner.end_lsn.unwrap_or(Lsn(u64::MAX));
+        format!("inmem-{:016X}-{:016X}", self.start_lsn.0, end_lsn.0)
+    }
```

Comment:
This range concept is high risk for subtle off-by-ones, would prefer to use the single fn above to get it.

```suggestion
    pub fn short_id(&self) -> String {
        let range = self.get_current_lsn_range();

        format!("inmem-{:016X}-{:016X}", range.start.0, range.end.0)
    }
```